### PR TITLE
break ci into files to use paths: so we can skip unneeded test runs

### PR DESCRIPTION
**Pull request by jakecoffman**
Dependabot rebases PRs and kicks off every CI job so we're running out of Actions runners. This reduces the amount of CI jobs run by only running ones that are applicable to the change.

### CODE REVIEW

**Review by landongrindheim — DISMISSED** [review]
I ❤️ this change!

**Comment by jakecoffman** [comment]
Unfortunately I think we're going to have to remove the required checks for the individual CI runs for this to work. I thought about making `ci-bundler-skip.yml` that did a `paths-ignore: [bundler]`, but if you edit `bundler/` plus common it would still run the skip I think 🤔

File: .github/workflows/ci-bundler.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: CI Bundler
+on:
+  push:
+    branches:
+      - "main"
+    paths:
+      - Dockerfile
+      - common
+      - bundler
```
Comment (landongrindheim) [inline .github/workflows/ci-bundler.yml:9]: I wanted to highlight to anyone concerned with this change that _all_ workflows will run when either `common/`  or `Dockerfile` change.

Comment (jakecoffman) [reply .github/workflows/ci-bundler.yml]: Assuming I can get the pattern right!

**Comment by jakecoffman** [comment]
closing for #5620

**Pull request closed without merging**